File: Unidad 1/T1.1 SIN BD/TI1.1 SOAP_JAVA_CONUNI_GR01/01 SERVIDOR/WS_CONUNI_SOAPJAVA_GRO01/src/main/java/ec/edu/monster/controlador/ConversionController.java
package ec.edu.monster.controlador;

import ec.edu.monster.modelo.ConversorLongitud;
import ec.edu.monster.modelo.ConversorMasa;
import ec.edu.monster.modelo.ConversorTemperatura;

public class ConversionController {

    private final ConversorLongitud longit = new ConversorLongitud();
    private final ConversorMasa masa = new ConversorMasa();
    private final ConversorTemperatura temp = new ConversorTemperatura();

    // LONGITUD
    public double kilometrosAMillas(double km) { return longit.kilometrosAMillas(km); }
    public double metrosAPies(double m)       { return longit.metrosAPies(m); }
    public double pulgadasACentimetros(double p) { return longit.pulgadasACentimetros(p); }

    // MASA
    public double kilogramosALibras(double kg) { return masa.kilogramosALibras(kg); }
    public double gramosAOnzas(double g)       { return masa.gramosAOnzas(g); }
    public double librasAKilogramos(double lb) { return masa.librasAKilogramos(lb); }

    // TEMPERATURA
    public double celsiusAFahrenheit(double c)   { return temp.celsiusAFahrenheit(c); }
    public double fahrenheitACelsius(double f)   { return temp.fahrenheitACelsius(f); }
    public double celsiusAKelvin(double c)       { return temp.celsiusAKelvin(c); }
}

File: Unidad 1/T1.1 SIN BD/TI1.1 SOAP_DOTNET_CONUNI_GR01/02 CLIMOV/app/src/main/java/com/example/a02climov/controller/ConversionController.kt
package com.example.a02climov.controller

import com.example.a02climov.service.SoapClient

enum class ConversionType(val method: String, val param: String) {
    KM_A_MILLAS("KilometrosAMillas", "km"),
    M_A_PIES("MetrosAPies", "m"),
    PULGADAS_A_CM("PulgadasACentimetros", "p"),

    KG_A_LB("KilogramosALibras", "kg"),
    G_A_OZ("GramosAOnzas", "g"),
    LB_A_KG("LibrasAKilogramos", "lb"),

    C_A_F("CelsiusAFahrenheit", "c"),
    F_A_C("FahrenheitACelsius", "f"),
    C_A_K("CelsiusAKelvin", "c");
}

object ConversionController {
    fun convert(type: ConversionType, value: Double): Double {
        return SoapClient.callDoubleMethod(type.method, type.param, value)
    }
}
